alua.c

```diff
@@ -295,8 +295,10 @@ int tcmu_get_alua_grps(struct tcmu_device *dev,
 	}
 
 	for (i = 0; i < n; i++) {
-		if (!strcmp(namelist[i]->d_name, "default_tg_pt_gp"))
-			continue;
+		if (!strcmp(namelist[i]->d_name, "default_tg_pt_gp")) {
+			if (failover_is_supported(dev))
+				continue;
+		}
```

Comment:
Did you see the comment at the top of tcmu_get_alua_grps? Older kernels are pretty nasty.

The problem is older kernels barf at various places when trying to use ALUA, so if we blindly scan we can end up hanging the entire kernel at the worst case. We assume some tool has done some check kernel check and when it is able to setup a group we are able to access it. But we recently added a kernel file so it now tells us if it is supported, but there are some kernels that did not have the file, but ALU A is supported, so I guess take your poison :)

But we could safely do your patch if we check for the alua_supported bits. Using the helper function to check if a file exists from that other PR check for the file:

/sys/kernel/config/target/core/user_N/devname/attrib/alua_support

if it exists and the value is 1 then the above chunk of code will be ok.

Note: for RHGS RHEL 7.4 does not have this file but I think all the needed alua fix ups are in it. RHEL 7.5 has everything so you would be ok for that.

Reply:
Yes, I saw that comments. So just one RFC here and asking for you advice here.

So I am thinking if we won't use the default alua, just create a new one, such as "glfstpg", will this be okay ?

Reply:
> So I am thinking if we won't use the default alua, just create a new one, such as "glfstpg", will this be okay ?

Yes.